int32 real_part
int32 complex_part
